### ISSUE TITLE
Auto upload does not work - Crash with `Parameter specified as non-null is null` (NPE)

### ISSUE THREAD

**Issue by MihailCosmin** [nextcloud/android#12586]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

1. Go to Settings > Auto upload
2. Select local Folder >  Configure
3. Enable Auto upload
4. Select Remote Folder - for example Photos
5. Click Choose
6. Back in the Auto Upload Menu the selected folder is not displayed as the Remote Folder and the app crashes with below error (see Android Logs)

### Expected behaviour

Selected folder should be displayed and Auto upload should work

### Actual behaviour

Selected folder does not display and Auto upload does not work

Without Auto upload the app is useless.... this  hasn't worked for months now....

### Android version

13

### Device brand and model

Samsung Galaxy Note 10 Lite

### Stock or custom OS?

Stock

### Nextcloud android app version

3.28.0

### Nextcloud server version

28.0.2

### Using a reverse proxy?

No

### Cause of error
```java
Exception in thread "main" java.lang.NullPointerException: Parameter specified as non-null is null: method com.owncloud.android.ui.dialog.SyncedFolderPreferencesDialogFragment$setupListeners$1$5.onItemSelected, parameter view
    at com.owncloud.android.ui.dialog.SyncedFolderPreferencesDialogFragment$setupListeners$1$5.onItemSelected(Unknown Source:2)
    at android.widget.AdapterView.fireOnSelected(AdapterView.java:1396)
    at android.widget.AdapterView.dispatchOnItemSelected(AdapterView.java:1385)
    at android.widget.AdapterView.selectionChanged(AdapterView.java:1368)
    at android.widget.AdapterView.checkSelectionChanged(AdapterView.java:1557)
    at android.widget.AdapterView.handleDataChanged(AdapterView.java:1533)
    at android.widget.AbsSpinner.onMeasure(AbsSpinner.java:193)
    at android.widget.Spinner.onMeasure(Spinner.java:645)
    at androidx.appcompat.widget.AppCompatSpinner.onMeasure(AppCompatSpinner.java:444)
    at android.view.View.measure(View.java:27731)
    at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7352)
    at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1552)
    at android.widget.LinearLayout.measureVertical(LinearLayout.java:842)
    at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
    at android.view.View.measure(View.java:27731)
    at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7352)
    at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1552)
    at android.widget.LinearLayout.measureVertical(LinearLayout.java:842)
    at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
    at android.view.View.measure(View.java:27731)
    at android.widget.ScrollView.measureChildWithMargins(ScrollView.java:2352)
    at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
    at android.widget.ScrollView.onMeasure(ScrollView.java:699)
    at android.view.View.measure(View.java:27731)
    at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7352)
    at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1552)
    at android.widget.LinearLayout.measureVertical(LinearLayout.java:842)
    at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
    at android.view.View.measure(View.java:27731)
    at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7352)
    at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
    at android.view.View.measure(View.java:27731)
    at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7352)
    at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
    at android.view.View.measure(View.java:27731)
    at androidx.appcompat.widget.AlertDialogLayout.tryOnMeasure(AlertDialogLayout.java:134)
    at androidx.appcompat.widget.AlertDialogLayout.onMeasure(AlertDialogLayout.java:64)
    at android.view.View.measure(View.java:27731)
    at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7352)
    at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
    at androidx.appcompat.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:145)
    at android.view.View.measure(View.java:27731)
    at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7352)
    at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
    at android.view.View.measure(View.java:27731)
    at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7352)
    at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
    at android.view.View.measure(View.java:27731)
    at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7352)
    at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1552)
    at android.widget.LinearLayout.measureVertical(LinearLayout.java:842)
    at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
    at android.view.View.measure(View.java:27731)
    at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7352)
    at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
    at com.android.internal.policy.DecorView.onMeasure(DecorView.java:1379)
    at android.view.View.measure(View.java:27731)
    at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:4841)
    at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:3397)
    at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3732)
    at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:3116)
    at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:10885)
    at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1301)
    at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1309)
    at android.view.Choreographer.doCallbacks(Choreographer.java:923)
    at android.view.Choreographer.doFrame(Choreographer.java:852)
    at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1283)
    at android.os.Handler.handleCallback(Handler.java:942)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loopOnce(Looper.java:226)
    at android.os.Looper.loop(Looper.java:313)
    at android.app.ActivityThread.main(ActivityThread.java:8762)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:604)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30280090`
* Build flavor: `gplay`

### Device information
* Brand: `samsung`
* Device: `r7`
* Model: `SM-N770F`
* Id: `TP1A.220624.014`
* Product: `r7naeea`

### Firmware
* SDK: `33`
* Release: `13`
* Incremental: `N770FXXS9HWJ1`

**Comment by awptechnologies** [comment]
I am also having this problem as well as the server not available error. If i log in through web browser i can see all of my files. The error only happens on the android app

**Comment by PYCODR** [comment]
same here, server not available @Android 14, Client App 3.28.0, NextCloudPi 26.0.3

**Comment by awptechnologies** [comment]
im on 28.0.2 and issue still exists

**Comment by antmarra** [comment]
Hi everybody,
I'm on Samsung Galaxy S23 Ultra with Android 14 and OneUI 6.1
Auto upload of my photos doesn't work anymore since last 25 March (it has worked flawlessly since 3/25).
I've tried to disable and re-enable auto upload of DCIM/Camera local folder, I've tried to move the missing photos to another folder and then to move them back to the original folder, but to no avail.
Then, I've downloaded and installed the 3.28.1 version from GitHub, but again it doesn't work.
Moreover, the new installed version was reported as 3.28.0 in the App Info on the phone...
Please help! Thanks

**Comment by joshtrichards** [comment]
May be fixed by #13070 / #13102

Please check in >=v3.29.1
